packages/expansion-panel/src/components/expansion-panel/index.tsx

```diff
@@ -0,0 +1,97 @@
+import React, {
+  FC,
+  isValidElement,
+  Children,
+  useState,
+  useEffect,
+  HTMLAttributes,
+  ComponentType
+} from 'react';
+
+import ExpansionPanelHead from '../expansion-panel-head';
+import ExpansionPanelBody from '../expansion-panel-body';
+
+import SC from './styled';
+
+export interface ExpansionIndicator {
+  expand: JSX.Element;
+  collapse: JSX.Element;
+}
+
+export interface Props extends HTMLAttributes<HTMLElement> {
+  /**
+   * A flag indicating whether extension panel is expanded
+   * @type {boolean}
+   * @default false
+   */
+  isExpanded?: boolean;
+  /**
+   * A flag indicating whether extension panel should expand on head click
+   * @type {boolean}
+   * @default true
+   */
+  shouldExpandOnHeadClick?: boolean;
+  /**
+   * Override expansion indicator
+   * @type {ExpansionIndicator}
+   */
+  expansionIndicator?: ExpansionIndicator;
+  /**
+   * Component to render
+   * @default div
+   */
+  as?: keyof JSX.IntrinsicElements | ComponentType<any>;
+}
+
+export const ExpansionPanel: FC<Props> = ({
+  isExpanded: isExpandedProp = false,
+  shouldExpandOnHeadClick = true,
+  expansionIndicator,
+  onClick,
+  children,
+  ...props
+}) => {
+  const [isExpanded, setIsExpanded] = useState(isExpandedProp);
+
+  const toggleExpansion = () => setIsExpanded(!isExpanded);
+
+  useEffect(() => setIsExpanded(isExpandedProp), [isExpandedProp]);
+
+  const renderExpansionPanelHeadChildren = () =>
+    Children.map(children, child =>
+      isValidElement(child) && child.type === ExpansionPanelHead ? child : null
+    )?.shift();
+
+  const renderExpansionPanelBodyChildren = () =>
+    Children.map(children, child =>
+      isValidElement(child) && child.type === ExpansionPanelBody ? child : null
+    )?.shift();
```

Comment:
Deler av koden kan flyttes til `utils.ts`. Se gjerne https://github.com/fellesdatakatalog/fdk-kit/blob/master/packages/tabs/src/components/tabs/utils.ts for inspirasjonen